### PR TITLE
Scattered ideas for the paper narrative

### PR DESCRIPTION
**Pull request by marcharper**
@drvinceknight

### CODE REVIEW

**Comment by marcharper** [comment]
I guess we should include a plot like the multi-player one in this [notebook](https://github.com/Axelrod-Python/Axelrod-notebooks/blob/master/Moran%20Process.ipynb) to illustrate the libraries' abilities. We could also compute a stationary distribution for a nonzero mutation example (or save it for the future).

File: tex/main.tex [hunk 1]
```diff
@@ -28,6 +28,45 @@ \section{Introduction}\label{sec:introduction}
...
+
+\[ \rho = \frac{1 - r^{-i}}{1 - r^{-N}}\]
+and $\rho = 1 / N$ if $r=1$ (the neutral fixation probability).
+
+This corresponds to a game matrix [[1, 1], [r, r]] (or [[r, r], [1, 1]]), which is of course not what we have -- it's a little complicated because our "fitness" is not the payout from the game matrix, rather the sum of the total scores of all the interactions each round. So ALLC and TFT are neutral wrt to each other because they will have the same score each round, giving an effective fitness landscape $f(i, N-i) = A [i, N-i]^T$ given by the matrix $A = [[1,1],[1,1]]$. This means that noise and the number of turns per Moran round are significant parameters. I think we should fix the turns at 200; some recent authors run the turns to infinity (to reach stationarity on the sub-"Markov process" on the states (C, C), (C, D), (D, C), (D, D)) but we can't analytically compute the stationary distribution for strategies that use more than one round of memory (and it's not really a Markov process for more than one round of memory anyway). Plus it's unrealistic, and ultimately just amounts to a transform of the game matrix.
```
Comment (drvinceknight) [inline tex/main.tex:36]: Cool, I'm currently running up until N=12 with 100 turns but it's little effort for me to up that to 200 or even 500?

File: tex/main.tex [hunk 2]
```diff
@@ -28,6 +28,45 @@ \section{Introduction}\label{sec:introduction}
...
+
+To see if one strategy is not neutral with respect to another, we want to empirically measure the fixation probability and compare to the neutral rate. To do this right we need a lot of counts, since we're estimating a binomial probability p with variance $p(1-p) / k$ and $p$ is close to $1 / N$. To get the variance small you need something like $k>1000$ observations ( we can work out the precise requirements).
+
+Note we're not estimating $r$ for each strategy (pair) since we're in a frequency dependent situation, so we need to look at the population states (1, N-1) and (N-1, 1) for every pair of strategies, i.e. we can't assume that we're in a $\rho \leftrightarrow 1-\rho$ symmetry. More precisely, $\rho_{(1, N-1)} != 1 - \rho_{(N-1, 1)}$ in general. However we can (for fun) compute $r$ from $\rho$ with Newton's method (it's not easily invertable for $N > 3$), or take a Bayesian approach on what the distribution of $\rho$ is and then compute a distribution for $r$ in the usual way.
```
Comment (drvinceknight) [inline tex/main.tex]: Currently running with 1000 repetitions. Could potentially double that, once we see how long it takes to run etc.

File: tex/main.tex [hunk 3]
```diff
@@ -28,6 +28,45 @@ \section{Introduction}\label{sec:introduction}
...
+
+To see if one strategy is not neutral with respect to another, we want to empirically measure the fixation probability and compare to the neutral rate. To do this right we need a lot of counts, since we're estimating a binomial probability p with variance $p(1-p) / k$ and $p$ is close to $1 / N$. To get the variance small you need something like $k>1000$ observations ( we can work out the precise requirements).
```
Comment (drvinceknight) [inline tex/main.tex:38]: Currently running with 1000 repetitions. Could potentially double that, once we see how long it takes to run etc.

File: tex/main.tex [hunk 4]
```diff
@@ -28,6 +28,45 @@ \section{Introduction}\label{sec:introduction}
...
+as a function of N across the entire population of strategies. This can really open up if you want to say optimize a parameterized strategy to be most resistant to invasion (a topic of future work, perhaps) -- for example Random(p) for what p is best?
+
+The existing notebook attempts to get at 1 and 2 by looking at the distributions of fixation probabilities for each strategy -- that's what the box plots for each N try to visualize for particular N, and the "Player Rankings by Median vs. Population Size" for how the cooperative strategies become more successful as N increases. That plot is the main takeaway IMO, and reinforces the "evolution of cooperation" narrative that's so popular. We can tie back to Press and Dyson here -- yes, ZD strategies are good Head-to-Head and in small populations, but they aren't great when the population size gets bigger. How much bigger? Even at N=4 there is a dramatic decline for ZD-extort. Note that this goes against the claims of Stewart and Plotkin (they claimed that ZD strategies basically dominate the Moran process no matter how much memory you allow). This also matches our tournament results -- ZD strategies win matches but not tournaments.
```
Comment (drvinceknight) [inline tex/main.tex:51]: That's interesting. Stewart is coming to give a talk at Cardiff in December. I'll talk to him about this.

File: tex/main.tex [hunk 5]
```diff
@@ -28,6 +28,45 @@ \section{Introduction}\label{sec:introduction}
...
+
+It would be great to see how the ensemble strategies (meta strategies) fare, if we don't mind burning the CPU cycles. I left them out of my initial analysis.
```
Comment (drvinceknight) [inline tex/main.tex:53]: Let's see how this initial re run of the data goes?

File: tex/main.tex [hunk 6]
```diff
@@ -36,6 +75,8 @@ \section{Introduction}\label{sec:introduction}
     \item Short discussion about the Axelrod library.
 \end{itemize}
 
+I'm happy to write this section. We can lift some references from one of my papers on the Moran process.
```
Comment (drvinceknight) [inline tex/main.tex:78]: Make sense, I kind of threw this (and those two citations) just to get the structure up :)

File: tex/main.tex (hunk 1 above)
Comment (marcharper) [reply tex/main.tex:36]: Some strategies are hard-coded to expect 200, but 100 is probably fine for 95+%.

**Pull request merged**